pkg/kubelet/kuberuntime/kuberuntime_manager_test.go

```diff
@@ -955,6 +955,33 @@ func TestComputePodActions(t *testing.T) {
 	}
 }
 
+func TestKillAllContainersIgnoringFailures(t *testing.T) {
+	_, _, m, err := createTestRuntimeManager()
+	require.NoError(t, err)
+
+	// Creating a pair reference pod and status for the test cases to refer
+	// the specific fields.
+	pod, status := makeBasePodAndStatus()
+	pod.Spec.RestartPolicy = v1.RestartPolicyNever
+	status.ContainerStatuses[1].State = kubecontainer.ContainerStateUnknown
+	actions := podActions{
+		SandboxID:         status.SandboxStatuses[0].Id,
+		ContainersToKill:  getKillMap(pod, status, []int{0, 1}),
+		ContainersToStart: []int{1},
+	}
```

Comment:
Could you adds comments to each of the lines that sets up this test pod and why it is set up in that way.   It isn't immediately apparent what about this pod results in an error in `killContainer()`.

Reply:
@sjenning 
I added some comment.
Please take a look.

Reply:
@sjenning 
Please take a look.